### ISSUE TITLE
Feature request: SDL Gradient

### ISSUE THREAD

**Issue by Jocker666z** [karlstav/cava#431]
I tinkered with this, if it helps, I think it's better written and as it stands it's hardcoded.

![cava](https://user-images.githubusercontent.com/16121204/151706949-c777be4f-57c5-43de-b356-204038d87a00.png)

in sdl_cava.c:
```
            // draw new bar
            fillRect.x = bar * (bar_width + bar_spacing) + remainder;
            fillRect.y = height - ( bars[bar] / 5 );
            fillRect.w = bar_width;
            fillRect.h = bars[bar] / 5;
            SDL_SetRenderDrawColor(gRenderer, 50, 50, 50, 0xFF);
            SDL_RenderFillRect(gRenderer, &fillRect);
            fillRect.x = bar * (bar_width + bar_spacing) + remainder;
            fillRect.y = height - ( ( bars[bar] / 5 ) * 2 );
            fillRect.w = bar_width;
            fillRect.h = bars[bar] / 5;
            SDL_SetRenderDrawColor(gRenderer, 75, 75, 75, 0xFF);
            SDL_RenderFillRect(gRenderer, &fillRect);
            fillRect.x = bar * (bar_width + bar_spacing) + remainder;
            fillRect.y = height - ( ( bars[bar] / 5 ) * 3 );
            fillRect.w = bar_width;
            fillRect.h = bars[bar] / 5;
            SDL_SetRenderDrawColor(gRenderer, 100, 100, 100, 0xFF);
            SDL_RenderFillRect(gRenderer, &fillRect);
            fillRect.x = bar * (bar_width + bar_spacing) + remainder;
            fillRect.y = height - ( ( bars[bar] / 5 ) * 4 );
            fillRect.w = bar_width;
            fillRect.h = bars[bar] / 5;
            SDL_SetRenderDrawColor(gRenderer, 125, 125, 125, 0xFF);
            SDL_RenderFillRect(gRenderer, &fillRect);
            fillRect.x = bar * (bar_width + bar_spacing) + remainder;
            fillRect.y = height - ( ( bars[bar] / 5 ) * 5 );
            fillRect.w = bar_width;
            fillRect.h = bars[bar] / 5;
            SDL_SetRenderDrawColor(gRenderer, 150, 150, 150, 0xFF);
            SDL_RenderFillRect(gRenderer, &fillRect);
```

**Comment by karlstav** [comment]
I have considered gradient option in sdl, but haven't found any easy way doing it yet. I think it must be done manually, like discussed here:

https://stackoverflow.com/questions/20348616/how-to-create-a-color-gradient-in-sdl/42234816

**Comment by karlstav** [comment]
since we already have sdl though, it would be more fun to do this on the gpu side.

**Comment by SpyderMe** [comment]
When I enable the gradient option in the config and have colors defined, I get no bars.  Yet, everything works fine without the gradient option enabled and I get nice solid color bars.  I can readily change the foreground and background colors and was interested in playing around with the gradient options, but it seems as though I am missing something.  I have reviewed a handful of articles, but maybe everyone else is using a different terminal.  I'm booting to the terminal/console on a later/latest Raspian/Bullseye release and running CAVA via a startup script.  Any help is appreciated.  Thanks.

**Comment by karlstav** [comment]
@SpyderMe it sounds like you are on the wrong ticket. If you are in the console, as in no desktop, then you are not using SDL. However gradient is not supported in the console, so there might be an issue here where cava should have given an error and exited instead of showing no bars.

**Comment by nikp123** [comment]
Just to add a bit of material if it's needed, I already made one for my fork ages ago and it worked fine, but I've deleted it as I've switched to OpenGL for my rendering. The code is still under a tag on my projects GitHub: https://github.com/nikp123/xava/blob/271f4b76693cb2463a8185944168d6d1ada516bb/src/output/graphical_sdl.c#L143

It's a fully smooth gradient (hence the quite complex code and not the prettiest) but not sure if your choice of gradient was stylistic or a bug. I'll try to find a screenshot.
